加入“优化占用空间”部分 (#45)
spencerwooo/dowww: docs/dev/2-CLI/2-3-Others.md

![](https://i.loli.net/2020/01/04/DZfJjm9PUhxl1AI.png)

> 

Reviewer: ```suggestion

### 优化 WSL 2 虚拟磁盘占用空间

```

![](https://i.loli.net/2020/01/04/DZfJjm9PUhxl1AI.png)

WSL 2 使用 [VHDX](https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/boot-to-vhd--native-boot--add-a-virtual-hard-disk-to-the-boot-menu) 格式的虚拟磁盘储存文件，磁盘大小会在需要时自动扩容，长时间使用可能会占用较大的磁盘空间。我们可以通过 [`Optimize-VHD`](https://docs.microsoft.com/en-us/powershell/module/hyper-v/optimize-vhd?view=win10-ps) 命令来优化其占用空间。

该命令是 Hyper-V 功能的一部分，我们需要在 [Windows 功能](https://jingyan.baidu.com/article/a378c960cb5b39b328283092.html)中启用 **Hyper-V » Hyper-V 管理工具 » Windows PowerShell 的 Hyper-V 模块** 以及 **Hyper-V » Hyper-V 平台 » Hyper-V 服务**：

> ![](https://i.loli.net/2020/02/29/u96cXs3QR7DFIew.png)

Reviewer: ```suggestion
![](https://i.loli.net/2020/02/29/A2l1zHDUWQr6NL8.png)
```

![](https://i.loli.net/2020/01/04/DZfJjm9PUhxl1AI.png)

WSL 命令行方面的配置、工具、操作和问题异常处理等内容基本介绍完毕。接下来的一章，我们将对利用 Visual Studio Code 和 WSL 配合进行工作开发的内容进行介绍。

### 优化占用空间

WSL 2 使用 [VHDX](https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/boot-to-vhd--native-boot--add-a-virtual-hard-disk-to-the-boot-menu) 格式的虚拟磁盘储存文件，磁盘大小会在需要时自动扩容，长时间使用可能会占用较大的磁盘空间。我们可以通过 [`Optimize-VHD`](https://docs.microsoft.com/en-us/powershell/module/hyper-v/optimize-vhd?view=win10-ps) 命令来优化其占用空间。

> 该命令是 Hyper-V 功能的一部分，我们需要确保 **Hyper-V/Hyper-V 管理工具/Windows PowerShell 的 Hyper-V 模块**及**Hyper-V/Hyper-V 平台/Hyper-V 服务**处于启用状态：

Reviewer: ```suggestion
该命令是 Hyper-V 功能的一部分，我们需要确保 `Hyper-V » Hyper-V 管理工具 » Windows PowerShell 的 Hyper-V 模块` 及 `Hyper-V » Hyper-V 平台 » Hyper-V 服务` 处于启用状态：
```

1. 这里的表述不太明确，我第一眼扫过去就没看明白这里代指的是子功能的结构
2. 这里希望也能稍微说一下如何找到「设置 Windows 功能」的页面

![](https://i.loli.net/2020/02/29/u96cXs3QR7DFIew.png)

> 然后以管理员权限执行 `Optimize-VHD -Path <VHDX 文件路径> -Mode Full` 即可。如果不清楚存储路径，可以运行注册表编辑器（`regedit`），在 `HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\CurrentVersion\Lxss\{随机 GUID}\BasePath` 下找到。

Reviewer: ```suggestion
然后以管理员权限执行 `Optimize-VHD -Path <VHDX 文件路径> -Mode Full` 即可。如果不清楚存储路径，可以运行注册表编辑器 Registry Editor（[How to open Registry Editor in Windows 10](https://support.microsoft.com/help/4027573/windows-10-open-registry-editor)），在 `HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\CurrentVersion\Lxss\{随机 GUID}\BasePath` 下找到。
```
